src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -200,6 +201,31 @@ abstract class AbstractServerTask extends AbstractTask {
         }
     }
 
+    protected void copyConfigDirectory() {
+        //merge default server.env with one in config directory
+        File configDirServerEnv = new File(server.configDirectory, "server.env")
+        if (configDirServerEnv.exists() && server.appendServerEnv) {
+            FileFilter fileFilter =   FileFilterUtils.notFileFilter(FileFilterUtils.nameFileFilter("server.env", null))
+            FileUtils.copyDirectory(server.configDirectory, getServerDir(project), fileFilter)
+
+            if(configDirServerEnv.exists()) {
```

Comment:
Just noticed that you don't need to check this again. It was already checked at line 207.